Tools/scripts/deepfreeze.py

```diff
@@ -109,11 +109,17 @@ def __init__(self, file: TextIO) -> None:
         self.cache: Dict[tuple[type, object, str], str] = {}
         self.hits, self.misses = 0, 0
         self.patchups: list[str] = []
+        self.deallocs: list[str] = []
         self.write('#include "Python.h"')
         self.write('#include "internal/pycore_gc.h"')
         self.write('#include "internal/pycore_code.h"')
         self.write('#include "internal/pycore_long.h"')
         self.write("")
+        with self.block("static void dealloc_codeobject(PyCodeObject co)", ";"):
```

Comment:
This should take a pointer, i.e., `PyCodeObject *co`! And the calls should use `&`.

Also, no trailing semicolon (but I'd like a blank line after it).

Also, maybe this should be inserted right before `_Py_Deepfreeze_Fini` instead of at the very top of the file.

Finally, maybe instead of a very long list of calls we could declare a NULL-terminated array of `PyCodeObject *` and just loop over that array in `_Py_Deepfreeze_Fini`. Such an array could be handy for other purposes as well.

Reply:
Fixed.

> Such an array could be handy for other purposes as well.

Will pass on for now, can be added if it is necessary for any use.